app/src/main/java/com/sopt/now/compose/SignUpActivity.kt

```diff
@@ -0,0 +1,218 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sopt.now.compose.Constants.Companion.MAX_ID_LENGTH
+import com.sopt.now.compose.Constants.Companion.MAX_PASSWORD_LENGTH
+import com.sopt.now.compose.Constants.Companion.MBTI_LENGTH
+import com.sopt.now.compose.Constants.Companion.MIN_ID_LENGTH
+import com.sopt.now.compose.Constants.Companion.MIN_PASSWORD_LENGTH
+import com.sopt.now.compose.Constants.Companion.USER_DATA
+import com.sopt.now.compose.ui.theme.NOWSOPTAndroidTheme
+
+class SignUpActivity : ComponentActivity() {
+    private lateinit var userData: SignUpData
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            NOWSOPTAndroidTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+                    showSignup(onSignupBtnClicked = { id, password, nickname, mbti ->
+                        userData = SignUpData(id, password, nickname, mbti)
+                        if (validateSignup()) {
+                            showToast(R.string.success_signup)
+                            navigateToLogin()
+                            finish()
+                        }
+                    })
+                }
+            }
+        }
+    }
+
+    private fun navigateToLogin() {
+        Intent(this, LoginActivity::class.java).apply {
+            putExtra(USER_DATA, userData)
+            setResult(RESULT_OK, this)
+        }
+    }
+
+    private fun validateSignup(): Boolean =
+        validateId() && validatePassword() && validateNickname() && validateMBTI()
+
+    private fun validateId(): Boolean {
+        require(userData.id.length in MIN_ID_LENGTH..MAX_ID_LENGTH) {
+            showToast(R.string.fail_sign_up_id)
+            return false
+        }
+        return true
+    }
+
+    private fun validatePassword(): Boolean {
+        require(userData.password.length in MIN_PASSWORD_LENGTH..MAX_PASSWORD_LENGTH) {
+            showToast(R.string.fail_sign_up_password)
+            return false
+        }
+        return true
+    }
+
+    private fun validateNickname(): Boolean {
+        require(userData.nickname.trim().isNotEmpty()) {
+            showToast(R.string.fail_sign_up_nickname)
+            return false
+        }
+        return true
+    }
+
+    private fun validateMBTI(): Boolean {
+        require(userData.mbti.length == MBTI_LENGTH) {
+            showToast(R.string.fail_sign_up_mbti)
+            return false
+        }
+        return true
+    }
+
+    private fun showToast(message: Int) =
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+}
+
+@Composable
+fun showSignup(
+    onSignupBtnClicked: (String, String, String, String) -> Unit
```

Comment:
함수의 매개변수에 SignUpData 타입으로 줘도 될것 같아요!
```suggestion
    onSignupBtnClicked: (SignUpData) -> Unit
```

Reply:
물론 이러면 아래 
onClick = { onSignupBtnClicked(id, password, nickname, mbti) },
에서 다시 초기화 해야하는 번거로움이 있겠네요..